panacea/web/static/js/app.js

```diff
@@ -1,104 +1,39 @@
 import 'babel-polyfill';
 import 'bootstrap';
 import 'whatwg-fetch';
-import * as Request from './request';
+import { RequestChain } from './request-chain';
 import * as View from './view';
 
-document.getElementById('file-form').addEventListener('submit', e => {
+View.fileForm.addEventListener('submit', e => {
   e.preventDefault();
-  submitFile.bind(e.target)();
+  const formData = new FormData(e.target);
+  submitFile(formData);
+  e.target.reset();
 });
 
-async function submitFile() {
+const submitFile = formData => {
   View.hideFileForm();
   View.hideResults();
   try {
-    await handleDrugsResponse(await Request.drugs(new FormData(this)));
+    retrievePML(async (pml) => {
+      const requestChain = new RequestChain(pml);
+      await requestChain.start(formData);
+      View.restoreFileForm();
```

Comment:
Just tried it, I don't think it will work. The finally block gets executed without waiting for the await in the `retrievePML`, it's a nuance of js' filthy callbacks and how our `retreivePML` function only calls back in a `fileReader`'s `onload` event handler.